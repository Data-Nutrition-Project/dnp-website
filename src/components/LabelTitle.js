import React from "react"

const LabelTitle = props => {
  return (
    <div className="Label-Title">
      <h1 className="dataset-label-header">{props.title}</h1>
      <span className="light-line">
        <h2 className="dataset-name-link"></h2>
      </span>
      <span className="weighted-line">
        <p className="dataset-origin-link"></p>
      </span>
    </div>
  )
}

// styles.module.css
// import {styles} from "./styles.module.css"
// styles.labelHeader
// className={styles.labelHeader}
export default LabelTitle
