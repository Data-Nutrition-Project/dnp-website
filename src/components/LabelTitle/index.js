import React from "react"

import PropTypes from "prop-types"

import styles from "./styles.module.css"

const LabelTitle = props => {
    return (
        <div className={styles.labelTitle}>
            <h1 className={styles.headerTitle}>Data Nutrition Label</h1>
            <div className={styles.pageSubHeaderRow}>
                <h2 className={styles.datasetLabelHeader}>
                    {props.datasetName}
                </h2>
                <a className={styles.iconLink} href={props.datasetNameLink}>
                    <img
                        className={styles.iconLinkImg}
                        src={require("../../images/linkimg.png")}
                        alt="icon for link"
                        id="datasetLink"
                    />
                </a>
            </div>
            <div className={styles.link}>
                <p className={styles.datasetOriginLink}>{props.datasetOrg}</p>
            </div>
            <div className={styles.link}>
                <a className={styles.datasetLink} href={props.datasetOrgLink}>
                    <img src="/linkimg.png" alt="" id="datasetLink" />
                </a>
            </div>
            <span className={styles.weightedLine}> </span>
        </div>
    )
}

LabelTitle.propTypes = {
    datasetName: PropTypes.string.isRequired,
    datasetOrg: PropTypes.string.isRequired,
    datasetNameLink: PropTypes.string.isRequired,
    datasetOrgLink: PropTypes.string.isRequired
}

LabelTitle.defaultProps = {
    datasetName: 'College Scoreboard Dataset 2020',
    datasetOrg: 'US Department of Education',
    datasetNameLink: '#',
    datasetOrgLink: '#'
}

export default LabelTitle
