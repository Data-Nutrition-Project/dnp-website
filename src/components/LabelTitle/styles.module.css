* {
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.headertTitle {
  margin: 0rem 0rem 0rem 0rem;
  width: 17.8125rem;
  height: 1.625rem;
  font-size: 1.625rem;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: normal;
  text-align: left;
}

.datasetLabel {
  margin: 0.6875rem 0rem 0rem 0rem;
  width: 28.125rem;
  height: 1.75rem;
  font-size: 1.75rem;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 0.79;
  letter-spacing: normal;
  text-align: left;
}

.iconLinkImg {
  margin-top: -1.5625rem;
  margin-left: 15rem;
  width: 0.875rem;
  height: 0.875rem;
  opacity: 1.2;
}

.weightedLine {
  margin: -5rem 0rem 0rem 0rem;
  border-bottom: 0.3125rem solid currentColor;
  display: inline-block;
  line-height: 1;
  width: 100%;
}

.datasetOriginLink {
  margin: -0.9375rem 0.3125rem 0rem 0rem;
  width: 12.75rem;
  height: 1rem;
  font-size: 1rem;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 0.88;
  letter-spacing: normal;
  width: 50%;
}

.link {
  display: flex;
  width: 100%;
}
