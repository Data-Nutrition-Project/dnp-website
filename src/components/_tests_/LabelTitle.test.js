import React from "react"
import renderer from "react-test-renderer"
import LabelTitle from "../LabelTitle"

describe("LabelTitle", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LabelTitle />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
