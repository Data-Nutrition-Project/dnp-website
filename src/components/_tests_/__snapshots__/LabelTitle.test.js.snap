// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelTitle renders correctly 1`] = `
<div
  className="labelTitle"
>
  <h1
    className="headerTitle"
  >
    Data Nutrition Label
  </h1>
  <div
    className="pageSubHeaderRow"
  >
    <h2
      className="datasetLabelHeader"
    >
      College Scoreboard Dataset 2020
    </h2>
    <a
      className="iconLink"
      href="#"
    >
      <img
        alt="icon for link"
        className="iconLinkImg"
        id="datasetLink"
        src="linkimg.png"
      />
    </a>
  </div>
  <div
    className="link"
  >
    <p
      className="datasetOriginLink"
    >
      US Department of Education
    </p>
  </div>
  <div
    className="link"
  >
    <a
      className="datasetLink"
      href="#"
    >
      <img
        alt=""
        id="datasetLink"
        src="/linkimg.png"
      />
    </a>
  </div>
  <span
    className="weightedLine"
  >
     
  </span>
</div>
`;
